common/meta/types/src/user_grant.rs

```diff
@@ -43,11 +53,14 @@ impl GrantObject {
         }
     }
 
-    /// Check if there's any privilege which can not be granted to this GrantObject
-    pub fn validate_privileges(&self, privileges: UserPrivilegeSet) -> Result<()> {
+    /// Check if there's any privilege which can not be granted to this GrantObject.
+    /// Some global privileges can not be granted to a database or table, for example,
+    /// a KILL statement is meaningless for a table.
+    pub fn validate_available_privileges(&self, privileges: UserPrivilegeSet) -> Result<()> {
```

Comment:
validate_available_privileges -> validate_grant_privileges?

Reply:
good idea, btw i think this function can be moved to interpreter_grant.rs, because it's the only call point, and GrantEntry is a low level object, do not really knows about the GRANT statement

Reply:
Ok, I guess many access checking can be add to interpreter, for access checking methods,  we already have a 'access' fold there

Reply:
oh, this function is not an access check but an argument validation.

each GrantObject has a set of available privileges, like a privilege about KILL is meaning less to a Table but available to Global, this function does this check.